packages/core/src/components/axes/toolbar.ts

```diff
@@ -0,0 +1,489 @@
+					.attr("y", 0)
+					.attr("width", buttonSize)
+					.attr("height", buttonSize);
+
+				const buttonIcon = DOMUtils.appendOrSelect(
+					buttonContainer,
+					"svg.toolbar-button-icon"
+				)
+					.attr(
+						"x",
+						buttonXPosition + Configuration.toolbar.iconPadding
+					)
+					.attr("y", Configuration.toolbar.iconPadding)
+					.attr("width", Configuration.toolbar.iconSize)
+					.attr("height", Configuration.toolbar.iconSize)
+					.attr("viewBox", "0 0 32 32")
+					.attr("role", Roles.IMG);
+
+				buttonIcon.html(button.iconSVGContent);
+				if (button.shouldBeDisabled()) {
+					buttonContainer
+						.classed("toolbar-button--disabled", true)
+						.classed("toolbar-button--focusable", false)
+						.attr("tabindex", -1)
+						.attr("role", null);
+					buttonIcon.classed("toolbar-button-icon--disabled", true);
+					buttonBackground.classed(
+						"toolbar-button-background--disabled",
+						true
+					);
+					buttonContainer.on("click", null).on("keyup", null);
+				} else {
+					buttonContainer
+						.classed("toolbar-button--disabled", false)
+						.classed("toolbar-button--focusable", true)
+						.attr("tabindex", 0)
+						.attr("role", Roles.BUTTON);
+					buttonIcon.classed("toolbar-button-icon--disabled", false);
+					buttonBackground.classed(
+						"toolbar-button-background--disabled",
+						false
+					);
+					buttonContainer
+						.on("click", button.clickFunction)
+						.on("keyup", () => {
+							if (
+								(event.key && event.key === "Enter") ||
+								event.key === " "
+							) {
+								event.preventDefault();
+
+								button.clickFunction();
+							}
+						});
+				}
+				buttonXPosition += buttonSize;
+			});
+
+			this.overflowButton = this.getContainerSVG().select(
+				"svg.toolbar-overflow-menu"
+			);
+
+			if (this.isOverflowMenuOpen()) {
+				// keep overflow menu displayed
+				this.showOverflowMenu(true);
+			}
+		}
+	}
+
+	isOverflowMenuOpen() {
+		return (
+			this.overflowMenu
+				.selectAll("ul.bx--overflow-menu-options--open")
+				.size() > 0
+		);
+	}
+
+	// show/hide overflow menu
+	showOverflowMenu(show: boolean) {
+		// update overflow button background
+		if (this.overflowButton) {
+			this.overflowButton.classed("toolbar-button--hovered", show);
+		}
+		if (show) {
+			this.services.events.dispatchEvent(
+				Events.Toolbar.SHOW_OVERFLOW_MENU
+			);
+		} else {
+			this.services.events.dispatchEvent(
+				Events.Toolbar.HIDE_OVERFLOW_MENU
+			);
+		}
+	}
+
+	focusOnPreviousEnabledMenuItem(currentItemIndex) {
+		const overflowMenuItems = this.getOverflowMenuItems();
+		let previousItemIndex = overflowMenuItems.length;
+		for (let i = currentItemIndex - 1; i >= 0; i--) {
+			const previousOverflowMenuItem = overflowMenuItems[i];
+			if (!previousOverflowMenuItem.shouldBeDisabled()) {
+				previousItemIndex = i;
+				break;
+			}
+		}
+		// only if previous enabled menu item found
+		if (previousItemIndex < overflowMenuItems.length) {
+			document
```

Comment:
can this be done with a d3 selection?

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/791/commits/70f749775ac447223e68fdf3e387b11e8cd07297.